### PR TITLE
VAULT-22103: Add docs for scan-docker-image command

### PR DESCRIPTION
**Pull request by prakashlinga**
## Description

Added a new command scan-docker-image to vault-radar CLI. This PR is to add documentation for this new command.

Resolves VAULT-22103

## How has this been tested?

Generated preview locally

### CODE REVIEW

**Review by vladimir-buzuev — COMMENTED** [review]
add the link to the file in `docs/README.md`?

File: docs/docker-image.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# `scan-docker-image` (experimental)
+This `vault-radar` command is used for scanning a docker image
+
+## Usage
+
+### Scanning a docker image
+
+Scan a public docker image (or a private image that is already pulled/dowloaded locally) and write the results to a file in CSV format, this is the default format for output. Image reference may optionally include a tag.[Docker engine](https://docs.docker.com/engine/install/) is a pre-requisite for scanning docker images using vault-radar.
```
Comment (kartheek-hc) [inline docs/docker-image.md:8]: Should we explicitly call out that we will attempt to scan latest tag if one isn't specified? Also, it isn't mandatory to have a latest tag

File: docs/docker-image.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+### Scanning a private docker image
+
+To scan a private docker image, specify the following environment variables to authenticate against the registry:
+1. `DOCKER_REGISTRY_USERNAME`
+2. `DOCKER_REGISTRY_PASSWORD`
+
+```bash
+vault-radar scan-docker-image -i <IMAGE REFERENCE> -f json -o <PATH TO OUTPUT>.csv
```
Comment (kartheek-hc) [inline docs/docker-image.md:21]: Can we remove `-f json` since output file has a csv extension?

Comment (prakashlinga) [reply docs/docker-image.md:21]: Updated to align with the example

**Review by kartheek-hc — APPROVED** [review]

**Review by vladimir-buzuev — APPROVED** [review]

**Pull request merged**